tests/test_update_picking.py

```diff
@@ -0,0 +1,503 @@
+                                               '%s for product Test product Apple' % picking.name,
+                             'No/Incorrect error message was thrown')
+
+    def test08_update_picking_repeated_serial_number_split(self):
+        """ Checks that the correct error is thrown when
+            when a serial number is repeated in a second call
+            to update_picking.
+        """
+        self.apple.write({'tracking': 'serial'})
+        create_info = [{'product': self.apple, 'qty': 2}]
+        picking = self.create_picking(self.picking_type_in,
+                                      origin='test_picking_origin',
+                                      products_info=create_info,
+                                      confirm=True)
+        products_info = [{
+                            'product_barcode': self.apple.barcode,
+                            'qty': 1,
+                            'serial_numbers': ['Apple0']
+                        }]
+        picking.update_picking(products_info=products_info)
+
+        with self.assertRaises(ValidationError) as e:
+            picking.update_picking(products_info=products_info)
+        self.assertEqual(e.exception.name, 'Serial numbers Apple0 already exist '
+                                             'in picking %s' % picking.name,
+                         'No/Incorrect error message was thrown')
+
+    def test09_update_picking_over_recived_single(self):
+        """ testing if over recived products in a single
+            call behaves as expected
+        """
+        create_info = [{'product': self.apple, 'qty': 4}]
+        picking = self.create_picking(self.picking_type_in,
+                                      origin='test_picking_origin',
+                                      products_info=create_info,
+                                      confirm=True)
+
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 5}]
+        picking.update_picking(products_info=products_info)
+        # check ordered_qty and qty_done  are as expected
+        self.assertEqual(picking.move_line_ids.ordered_qty, 4)
+        self.assertEqual(picking.move_line_ids.qty_done, 5)
+        picking.update_picking(validate=True)
+        self.assertEqual(picking.move_lines.state, 'done', 'Process doesn\'t complete')
+
+    def test10_update_picking_over_recived_split_lines(self):
+        """ testing if over recived products in seperate
+            calls behave as expected
+        """
+        create_info = [{'product': self.apple, 'qty': 4}]
+        picking = self.create_picking(self.picking_type_in,
+                                      origin='self_picking_origin',
+                                      products_info=create_info,
+                                      confirm=True)
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 4}]
+        picking.update_picking(products_info=products_info)
+        products_info[0]['qty'] =  2
```

Comment:
Use products_info_1 and _2? or completely override it: 
` products_info = [{'product_barcode': self.apple.barcode, 'qty': 2}]`